### DISCUSSION TITLE
Replacing parameters dynamically in language lines in Laravel 5.1

### DISCUSSION THREAD

**Discussion by essivision** [laravel/framework#10879] [General]
I have language file called `resources/lang/en/message.php` in laravel 5.1. Now I want to define a custom message line like this:

```
'deletemsg'=> 'The :name deleted successfully.',
```

and use in page.blade.php with following format:

```
{{trans('message.deletemsg')}}
```

In Laravel Documentation said to replace parameter it should be written:

```
{{ trans('messages.welcome', ['name' => 'Dayle']); }}
```

But I want the `:name` parameter dynamically load in all language and I don't type statically.In Laravel has a `validation.php` that has a attributes array to get variable of attribute. I like to do. I think we can do so again

**Comment by raualvron** [comment]
Hi @essivision 

I need exactly you asked in your post. Did you find a solution for it?

Thank you.

**Comment by icangku** [comment]
just change like this ['name' => $name];

**Comment by raualvron** [comment]
Hi @masterpieceicang 

Thank you for your answer but that doesn't work.

controller.php
```
return Redirect::route('auth.reset', ['token'=>$token])
                    ->with('error-message', trans('validation.password.no_update'));
```

validation.php (validation language lines)
```
'password'             => [
        'no_update' => 'Su :attribute no ha sido actualizada. Intentelo de nuevo.',
 ],

...

'attributes' => ['password' => 'contraseña'],
```

Expected result:
_Su contraseña no ha sido actualizada. Intentelo de nuevo._

I want the :attribute parameter dynamically change with the attribute I have set for all the languages

**Comment by joserick** [comment]
I have the same problem, any solution?

**Comment by maciekiwaniuk** [comment]
trans('messages.welcome', ['name' => $name]);

'welcome' => 'Wassssup :name!',

**Comment by soroushoracle** [comment]
Use  **__()** helper function  instead of _trans()_

**Reply by AlterwebStudio** [reply]
Nevertheless, the __() calls trans() method inside itself.
